web/frontend/src/components/slam-2d-render.vue

```diff
@@ -0,0 +1,121 @@
+<script setup lang="ts">
+
+import { threeInstance, resizeRendererToDisplaySize } from 'trzy';
+import { onMounted, onUnmounted, watch } from 'vue';
+import * as THREE from 'three';
+import { MapControls } from 'three/examples/jsm/controls/OrbitControls';
+import { PCDLoader } from 'three/examples/jsm/loaders/PCDLoader';
+import type { commonApi } from '@viamrobotics/sdk';
+
+interface Props {
+  name: string
+
+  /*
+   * NOTE: This is needed as vue doesn't support watchers for Uint8Array
+   * so we use the pointCloudUpdateCount as a signal that the pointcloud
+   * has changed & needs to be re rendered.
+   */
+  pointCloudUpdateCount: number
+  resources: commonApi.ResourceName.AsObject[]
+  pointcloud?: Uint8Array
+  pose?: commonApi.Pose
+}
+
+const props = defineProps<Props>();
+
+const loader = new PCDLoader();
+
+const container = $ref<HTMLElement>();
+
+const { scene, renderer } = threeInstance();
+
+const color = new THREE.Color(0xFF_FF_FF);
+renderer.setClearColor(color, 1);
+
+renderer.domElement.style.cssText = 'width:100%;height:100%;';
+
+const camera = new THREE.OrthographicCamera(-1, 1, 0.5, -0.5, -1, 1000);
+camera.userData.size = 2;
+
+const markerSize = 0.5;
+const marker = new THREE.Mesh(
+  new THREE.PlaneGeometry(markerSize, markerSize).rotateX(-Math.PI / 2),
+  new THREE.MeshBasicMaterial({ color: 'red' })
+);
+
+const controls = new MapControls(camera, renderer.domElement);
+
+const disposeScene = () => {
+  scene.traverse((object: THREE.Points | THREE.Material | unknown) => {
+    if (object instanceof THREE.Points) {
+      object.geometry.dispose();
+
+      if (object.material instanceof THREE.Material) {
+        object.material.dispose();
+      }
+    }
+  });
+
+  scene.clear();
+};
+
+const update = (pointcloud: Uint8Array, pose: commonApi.Pose) => {
+  const points = loader.parse(pointcloud.buffer, '');
+
+  const x = pose.getX!();
+  const y = pose.getY!();
+  marker.position.setX(x);
+  // NOTE: This is set to xz b/c we are projecting on the xz plane.
```

Comment:
because in the initial pr for parallel projection we went with XZ plane, so position endpoints need to match that assumption. In cartographer we rotated the getPosition endpoint to do this, orbslam did this by default(assuming a forward facing camera)

Reply:
@nicksanford , this is necessary so that GetPointCloudMap and GetPosition are using the same orientation, until we switch to GetPositionNew, which already rotates onto the XZ plane: https://github.com/viamrobotics/slam/blob/b293014dc772c91ce7c4a3dc014f8d558cc3d657/slam-libraries/viam-cartographer/src/slam_service/slam_service.cc#L69-L70.

Reply:
It would be helpful to add a TODO to change this when we switch GetPositionNew to GetPosition in https://viam.atlassian.net/browse/RSDK-1066

Reply:
I've updated [RSDK-1066](https://viam.atlassian.net/browse/RSDK-1066) & added a comment in this file for what I have understood needs to change when we integrate GetPositionNew into the frontend.